docs/assets/archi.blockdiag.txt

```diff
@@ -0,0 +1,33 @@
+blockdiag {
+  code_change -> run_linters -> run_reporters;
+  code_change -> languages;
+  code_change -> formatters
+  code_change -> tooling_linters;
+  run_linters -> console;
+  run_linters -> text;
+  run_linters -> json;
+  run_linters -> ide_config;
+  run_linters -> email;
+  run_linters -> file_io;
+  run_linters -> gh_pr;
+  run_linters -> gh_status;
+  run_linters -> tap;
+  run_linters -> updated_sources;
+
+  code_change [label = "Code Change"];
+  run_linters [label = "Run linters"];
+  run_reporters [label = "Run reporters"];
+  languages [label = "{language} languages", color="yellow"];
+  formatters [label = "{format} formatters", color="yellow"];
+  tooling_linters [label = "{tooling_format} tooling linters", color="yellow"];
+  console [label = "Console", color="greenyellow"];
+  text [label = "Text", color="greenyellow"];
+  json [label = "JSON", color="greenyellow"];
+  ide_config [label = "IDE Configuration", color="greenyellow"];
+  email [label = "E-Mail", color="greenyellow"];
```

Comment:
nit: `E-Mail` --> `Email` (as it was previously) since [`e-mail` is so 2000s](https://books.google.com/ngrams/graph?content=email%2Ce-mail&year_start=1800&year_end=2019&corpus=26&smoothing=3&direct_url=t1%3B%2Cemail%3B%2Cc0%3B.t1%3B%2Ce%20-%20mail%3B%2Cc0), and it has since grown up to become [a word of its own](https://books.google.com/ngrams/graph?content=email%2Cmail&year_start=1800&year_end=2019&corpus=26&smoothing=3&direct_url=t1%3B%2Cemail%3B%2Cc0%3B.t1%3B%2Cmail%3B%2Cc0).

Reply:
ok